src/main/java/com/example/demo/client/JsonPlaceholderWebClient.java

```diff
@@ -0,0 +1,135 @@
+    WebClient client;
+
+    public ClientResponse sendGet(String url, MethodTypeEnum query, HttpHeaders headers) {
+        WebClient.UriSpec<WebClient.RequestBodySpec> uriSpec = client.method(HttpMethod.GET);
+        WebClient.RequestBodySpec bodySpec = uriSpec.uri(ProjectVeriables.BASE_URL + url);
+        logger.info("Sending GET request to : " + url + "and getting element");
+        return Objects.requireNonNull(bodySpec.exchange().block(), "Response should be not null");
+    }
+
+    public ClientResponse sendPost(String url, MethodTypeEnum query, HttpHeaders headers, Object body) {
+        WebClient.UriSpec<WebClient.RequestBodySpec> uriSpec = client.method(HttpMethod.POST);
+        WebClient.RequestBodySpec bodySpec = uriSpec.uri(ProjectVeriables.BASE_URL + url);
+        WebClient.RequestHeadersSpec<?> headersSpec = bodySpec.bodyValue(body);
+        logger.info("Sending POST request to : " + url + " and creating new element");
+
+        return Objects.requireNonNull(headersSpec.exchange().block(), "Response should be not null");
+    }
+
+    public ClientResponse sendDelete(String url, MethodTypeEnum query, HttpHeaders headers) {
+        WebClient.UriSpec<WebClient.RequestBodySpec> uriSpec = client.method(HttpMethod.DELETE);
+        WebClient.RequestBodySpec bodySpec = uriSpec.uri(ProjectVeriables.BASE_URL + url);
+        logger.info("Sending DELETE request to : " + url + " and deleting element");
+        return Objects.requireNonNull(bodySpec.exchange().block(), "Response should be not null");
+    }
+
+    public ClientResponse sendPatch(String url, MethodTypeEnum query, HttpHeaders headers, Object body) {
+        WebClient.UriSpec<WebClient.RequestBodySpec> uriSpec = client.method(HttpMethod.PATCH);
+        WebClient.RequestBodySpec bodySpec = uriSpec.uri(ProjectVeriables.BASE_URL + url);
+        WebClient.RequestHeadersSpec<?> headersSpec = bodySpec.bodyValue(body);
+        return Objects.requireNonNull(headersSpec.exchange().block(), "Response should be not null");
+    }
+
+    public ClientResponse sendPut(String url, MethodTypeEnum query, HttpHeaders headers, Object body) {
+        WebClient.UriSpec<WebClient.RequestBodySpec> uriSpec = client.method(HttpMethod.PUT);
+        WebClient.RequestBodySpec bodySpec = uriSpec.uri(ProjectVeriables.BASE_URL + url);
+        WebClient.RequestHeadersSpec<?> headersSpec = bodySpec.bodyValue(body);
+        logger.info("Sending PUT request to : " + url  + " and changing element");
+        return Objects.requireNonNull(headersSpec.exchange().block(), "Response should be not null");
+    }
+
+    public Post getPost() {
+        ClientResponse clientResponse = sendGet("/posts/1", null, null);
+        return clientResponse.bodyToMono(Post.class).block();
+    }
+
+    public List<Post> postPost() {
+        Post newPostElement = new Post(9999, 12, "TEST TITLE", "TEST BODY");
+        ClientResponse clientResponse = sendPost("/posts/1", null, null, newPostElement);
```

Comment:
are you sure this is the correct endpoint?

Reply:
relevant endpoint is /posts. It is fixed